internal/server.go

```diff
@@ -481,6 +481,104 @@ func (s *Server) DiscoverSchemas(ctx context.Context, req *pub.DiscoverSchemasRe
 	}, err
 }
 
+func (s *Server) DiscoverRelatedEntities(ctx context.Context, req *pub.DiscoverRelatedEntitiesRequest) (*pub.DiscoverRelatedEntitiesResponse, error) {
+
+	db := s.session.DB
+	if db == nil {
+		return nil, errors.New("database connection not available")
+	}
+
+	var schemaNames []string
+	for _, schema := range req.GetToRelate() {
+		parts := strings.Split(schema.GetName(), ".")
```

Comment:
This should be based on the Id and not the Name as the user can change the Schema Name but not the Id.

You should also use the function DecomposeSafeName to decode the Schema Id into parts